src/modules/sharing/sharing.service.ts

```diff
@@ -1116,22 +1116,31 @@ export class SharingService {
 
     if (isUserPreCreated) {
       const encodedUserEmail = encodeURIComponent(userJoining.email);
-
-      new MailerService(this.configService)
-        .sendInvitationToSharingGuestEmail(
+      try {
+        await new MailerService(
+          this.configService,
+        ).sendInvitationToSharingGuestEmail(
           user.email,
           userJoining.email,
           item.plainName,
           {
-            signUpUrl: `${this.configService.get('clients.drive.web')}/new${
+            signUpUrl: `${this.configService.get(
+              'clients.drive.web',
+            )}/shared-guest?invitation=${
               createdInvite.id
-            }?invitation=${createdInvite.id}&email=${encodedUserEmail}`,
+            }&email=${encodedUserEmail}`,
             message: createInviteDto.notificationMessage || '',
           },
-        )
-        .catch(() => {
-          // no op
-        });
+        );
+      } catch (error) {
+        Logger.error(
+          `[SHARING/GUESTUSEREMAIL] Error sending email pre created userId: ${
+            userJoining.uuid
+          }, error: ${JSON.stringify(error)}`,
+        );
+        await this.removeInvite(user, createdInvite.id);
```

Comment:
If the guest user invitation fails, there is no point on persisting this invitation.